[atest] error C2090: Function returns array

```
expect_tomatch.cpp(83): error C2090: function returns array
commandline_option_test.cpp(706): note: see reference to class template instantiation 'atest::ExpectToMatch<T,char [1],atest::Matcher>' being compiled
        with
        [
            T=std::basic_string<char,std::char_traits<char>,std::allocator<char>>
        ]
```

C++ does not allow returning arrays from functions. Therefore following template code will fail the `V` is an array:

```
[[nodiscard]] auto evaluateExpression() const -> V
{
    if constexpr (std::is_invocable<T>::value)
    {
        return this->expression()();
    }
    else
    {
        return this->expression();
    }
}
```

--- a/projects/atest/expect_tomatch.cpp
+++ b/projects/atest/expect_tomatch.cpp
@@ -80,7 +80,7 @@ public:
     auto operator=(ExpectToMatch &&other) noexcept -> ExpectToMatch & = default;
 
 private:
-    [[nodiscard]] auto evaluateExpression() const -> V
+    [[nodiscard]] auto evaluateExpression() const -> auto
     {
         if constexpr (std::is_invocable<T>::value)
         {

--- a/projects/atest/test/expect_tobe_test.cpp
+++ b/projects/atest/test/expect_tobe_test.cpp
@@ -25,6 +25,10 @@ static const auto s = suite("Expect::toBe()", [] { //NOLINT(cert-err58-cpp)
         expect(1).toBe(2).toFail();
     });
 
+    test("match array", [] {
+        expect(std::string{}).toBe("");
+    });
+
     test("Callable is expected to fail but throws an exception (INTENTIONAL FAILURE)", [] {
         expect([]() -> int { throw std::runtime_error{"bad call"}; }).toBe(1).toFail();
     });

--- a/projects/atest/test/expect_tobe_test.cpp
+++ b/projects/atest/test/expect_tobe_test.cpp
@@ -25,7 +25,7 @@ static const auto s = suite("Expect::toBe()", [] { //NOLINT(cert-err58-cpp)
         expect(1).toBe(2).toFail();
     });
 
-    test("match array", [] {
+    test("Match array", [] {
         expect(std::string{}).toBe("");
     });
 

